diff --git a/test/e2e/service_to_service_test.go b/test/e2e/service_to_service_test.go
index ad93188b..7a98042a 100644
--- a/test/e2e/service_to_service_test.go
+++ b/test/e2e/service_to_service_test.go
@@ -139,6 +139,7 @@ func testProxyToHelloworld(t *testing.T, clients *test.Clients, helloworldDomain
 // The expected result is that the request sent to httpproxy app is successfully redirected
 // to helloworld app.
 func TestServiceToServiceCall(t *testing.T) {
+	t.Skip("Known to be broken in non-mesh cases.")
 	t.Parallel()
 	cancel := logstream.Start(t)
 	defer cancel()
@@ -185,6 +186,7 @@ func TestServiceToServiceCall(t *testing.T) {
 // Same test as TestServiceToServiceCall but before sending requests
 // we're waiting for target app to be scaled to zero
 func TestServiceToServiceCallFromZero(t *testing.T) {
+	t.Skip("Known to be broken in non-mesh cases.")
 	t.Parallel()
 	cancel := logstream.Start(t)
 	defer cancel()
