diff --git a/vendor/knative.dev/pkg/test/helpers/name.go b/vendor/knative.dev/pkg/test/helpers/name.go
index 7bf14a3bf..b2ba6c146 100644
--- a/vendor/knative.dev/pkg/test/helpers/name.go
+++ b/vendor/knative.dev/pkg/test/helpers/name.go
@@ -48,7 +48,11 @@ func ObjectPrefixForTest(t test.T) string {
 
 // ObjectNameForTest generates a random object name based on the test name.
 func ObjectNameForTest(t test.T) string {
-	return kmeta.ChildName(ObjectPrefixForTest(t), string(sep)+RandomString())
+	prefix := ObjectPrefixForTest(t)
+	if len(prefix) > 20 {
+		prefix = prefix[:20]
+	}
+	return kmeta.ChildName(prefix, string(sep)+RandomString())
 }
 
 // AppendRandomString will generate a random string that begins with prefix.
